### PR TITLE
ZCS-6075 GetAddressListInfo API constants, models, and util methods.

### PR DESCRIPTION
**Pull request by desouzas**
* Added constants, models, and ldap util methods for GetAddressListInfo - this API returns the meta info of the address list, and does not execute the query.

**Testing Done**
See jira ticket.

**Testing to be done by QA**
See testrail.

### CODE REVIEW

File: soap/src/java/com/zimbra/soap/admin/message/GetAddressListInfoResponse.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+     * @zm-api-field-description zimbra id of the address list
+     */
+    @XmlAttribute(name=AdminConstants.E_ID /* id */, required=true)
+    private String id;
+
+    /**
+     * @zm-api-field-tag name
+     * @zm-api-field-description name of the address list
+     */
+    @XmlElement(name=AdminConstants.E_NAME /* name */, required=false)
```
Comment (chinmay15) [inline soap/src/java/com/zimbra/soap/admin/message/GetAddressListInfoResponse.java]: required=true
Address list will always have a name

File: soap/src/java/com/zimbra/soap/admin/message/GetAddressListInfoResponse.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+     * @zm-api-field-description search filter for conditions
+     */
+    @XmlElement(name=AdminConstants.E_SEARCH_FILTER /* searchFilter */, required=false)
+    private EntrySearchFilterInfo searchFilter;
+
+    /**
+     * @zm-api-field-tag type
+     * @zm-api-field-description gal search type
+     */
+    @XmlAttribute(name=AdminConstants.A_TYPE /* type */, required=false)
```
Comment (chinmay15) [inline soap/src/java/com/zimbra/soap/admin/message/GetAddressListInfoResponse.java]: required=true
address list will always have gal search type. while creating address list if type is not provided, "all" is set by default. so while fetching address list, type will always be there.

File: soap/src/java/com/zimbra/soap/admin/message/GetAddressListInfoRequest.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+     * @zm-api-field-description name of the address list to get
+     */
+    @XmlElement(name=AdminConstants.E_NAME /* name */, required=false)
+    private String name;
+
+    /**
+     * @zm-api-field-tag domain
+     * @zm-api-field-description domain of the address list to get (needed with name)
+     */
+    @XmlElement(name=AdminConstants.E_DOMAIN, /* domain */ required=false)
```
[the remark this answers is not in the document]
Comment (desouzas) [reply soap/src/java/com/zimbra/soap/admin/message/GetAddressListInfoRequest.java]: You can have address lists with the same name in a different domain.

[the remark this answers is not in the document]
Comment (desouzas) [reply soap/src/java/com/zimbra/soap/admin/message/GetAddressListInfoRequest.java]: After discussion with @rupalid, this is resolved.

**Pull request merged**